src/plugins/agentDesktop/agentdesktop.ts

```diff
@@ -13,38 +15,95 @@ class AgentDesktopPlugin {
 
     }
     onHostCreate() {
-        let me:any = this;
+        let me: any = this;
         let cwInstance = me.hostInstance;
 
-        cwInstance.on("jwtGrantSuccess", (response:any) => {
+        cwInstance.on("jwtGrantSuccess", (response: any) => {
             if (!this.agentDesktopInfo) {
                 me.onInit();
-                this.config=me.extend(me,response)
+                this.config = me.extend(me, response)
                 //me.AgentDesktop(response.userInfo.userId, response);
                 this.agentDesktopInfo = new AgentDesktopPluginScript(this.config);
             }
         });
 
     }
     onInit() {
-        // let me = this;
+        let me: any = this;
         this.appendVideoAudioElemnts()
-        // this.koreCoBrowse = me.koreCoBrowseInit({});
-        // this.rrweb = me.rrwebInit({});
+
+        // agent connected and disconnected events
+        me.hostInstance.on('onWSMessage', (event: any) => {
+            if (event.messageData?.message?.type === 'agent_connected') {
+                localStorage.setItem("isAgentConnected", "true")
+            } else if (event.messageData?.message?.type === 'agent_disconnected') {
+                localStorage.setItem("isAgentConnected", "false")
+            }
+        });
+
+        // sent event style setting in user chat 
+        me.hostInstance.on('afterRenderMessage', (event: any) => {
+
+            if (localStorage.getItem("isAgentConnected") != "true") return;
+
+            if (event.msgData?.type === "currentUser") {
+                const msg = event.msgData.message;
+                if (!event.messageHtml.children('.sentIndicator').length) {
+                    event.messageHtml.append('<div class="sentIndicator">Sent</div>');
+
+                    me.hostInstance.bot.on('message', (message: any) => {
+                        var tempData = JSON.parse(message.data);
+                        if (!tempData) return;
+                        if (tempData.from === "bot" && tempData.type === "events" && tempData.message.clientMessageId === msg[0].clientMessageId) {
+
+                            if (tempData.message.type === "message_delivered") {
+                                if ($("#" + tempData.message.clientMessageId + " .sentIndicator").text() !== 'Read') {
```

Comment:
@KambagiriswamyBo-Kore  don't use the global selector with $ everywhere, use specific dom to search

Reply:
@RajasekharBa-Kore replaced global selector with instance selector.